aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,392 @@
+
+- Provides a central location for naming and configuring logical `HttpClient` instances. For example, a "github" client can be registered and pre-configured to access GitHub while a default client can be registered for other purposes.
+- Codifies the concept of outgoing middleware via delegating handlers in `HttpClient` and provides extensions for Polly-based middleware to take advantage of that. 
+- Manages the pooling and lifetime of underlying `HttpClientMessageHandler` instances to avoid common DNS problems that can be hit when managing `HttpClient` lifetimes manually.
+
+## Consumption patterns
+
+There are several ways that `HttpClientFactory` can be used in an app. None of them are strictly superior to another, it really depends on the constraints on the app.
+
+## Basic usage
+
+The `HttpClientFactory` can be used directly in code to access `HttpClient` instances. First, the services must be registered with the ServiceProvider.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient();
+    services.AddMvc();
+}
+```
+
+Once registered, code can accept a `IHttpClientFactory` anywhere services can be injected by the DI framework. The `IHttpClientFactory` can then be used to create a `HttpClient` instance.
+
+```csharp
+public class MyController : Controller
+{
+    IHttpClientFactory _httpClientFactory;
+
+    public MyController(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public IActionResult Index()
+    {
+        var client = _httpClientFactory.CreateClient();
+        var result = client.GetStringAsync("http://myurl/");
+        return View();
+    }
+}
+```
+
+Using `HttpClientFactory` like this is a good way to start refactoring an existing app, as it has no impact on the way `HttpClient` is used. In places where `HttpClient` instances are currenty created, replace those with a call to `CreateClient()`.
+
+## Named clients
+
+ If an app requires multiple distinct uses of `HttpClient`, each with different configurations, then an option is to use **named clients**. Any common configuration for a named `HttpClient` can be specified during registration.
+
+```csharp
+ public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient("github", c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // Github API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // Github requires a user-agent
+    });
+    services.AddHttpClient();
+}
+```
+
+Here `AddHttpClient` has been called twice, once with the name 'github' and once without. The GitHub-specific client has some default configuration applied, namely the base address and two headers required to work with the GitHub API.
```

Comment:
~~Here `AddHttpClient` has been called twice, once with the name 'github' and once without.~~
In the preceding code, `AddHttpClient` is called twice: once with the name "github" and once without.